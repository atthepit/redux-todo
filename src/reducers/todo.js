export const ADD_TODO = `ADD_TODO`
export const TOGGLE_TODO = `TOGGLE_TODO`

function todo(state = {}, action) {
  switch (action.type) {
  case ADD_TODO:
    return {
      id: action.id,
      text: action.text,
      complete: false,
    }
  case TOGGLE_TODO:
    if (action.id !== state.id)
      return state

    return {
      ...state,
      complete: !state.complete,
    }
  default:
    return state
  }
}

export default todo
