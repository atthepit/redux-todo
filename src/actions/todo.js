import { ADD_TODO, TOGGLE_TODO } from '../reducers/todo'

export const addTodo = (id, text) => ({
  type: ADD_TODO,
  id,
  text,
})

export const toggleTodo = (id) => ({
  type: TOGGLE_TODO,
  id,
})
