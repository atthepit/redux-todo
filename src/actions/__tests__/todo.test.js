import * as actions from '../todo'

describe('Todo action creators', () => {
  test('Add todo', () => {
    const id = 0
    const text = 'Hello world'

    expect(
      actions.addTodo(id, text)
    ).toMatchSnapshot()
  })

  test('Toggle todo', () => {
    const id = 0

    expect(
      actions.toggleTodo(id)
    ).toMatchSnapshot()
  })

})
