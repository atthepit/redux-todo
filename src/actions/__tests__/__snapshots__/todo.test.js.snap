exports[`Todo action creators Add todo 1`] = `
Object {
  "id": 0,
  "text": "Hello world",
  "type": "ADD_TODO",
}
`;

exports[`Todo action creators Toggle todo 1`] = `
Object {
  "id": 0,
  "type": "TOGGLE_TODO",
}
`;
